src/pudl_archiver/zenodo/api_client.py

```diff
@@ -56,35 +55,44 @@ class ZenodoDepositionInterface:
     def __init__(
         self,
         data_source_id: str,
+        downloader: AbstractDatasetArchiver,
         session: aiohttp.ClientSession,
         upload_key: str,
         publish_key: str,
-        api_root: str,
+        deposition_settings: Path,
+        create_new: bool = False,
         dry_run: bool = True,
+        sandbox: bool = True,
     ):
         """Prepare the ZenodoStorage interface.
 
+        Do not call this directly, use .create(). There's async stuff that needs to happen
+        for initialization.
+
         Args:
             data_source_id: Data source ID.
             session: Async http client session manager.
             upload_key: Zenodo API upload key.
             publish_key: Zenodo API publish key.
-            api_root: Zenodo API root URL. Points to sandbox or production API.
 
         Returns:
             ZenodoDepositionInterface
-
         """
-        self.testing = "sandbox" in api_root
+        if sandbox:
+            self.api_root = "https://sandbox.zenodo.org/api"
+        else:
+            self.api_root = "https://zenodo.org/api"
+
+        self.sandbox = sandbox
         self.data_source_id = data_source_id
 
-        self.api_root = api_root
         self.session = session
 
         self.upload_key = upload_key
         self.publish_key = publish_key
 
-        self.depositor = ZenodoDepositor(upload_key, publish_key, session, self.testing)
+        self.depositor = ZenodoDepositor(upload_key, publish_key, session, self.sandbox)
```

Comment:
We should initialize the Depositor elsewhere so that we remove the Zenodo dependencies in this class - I put a TODO in here in the rename PR.